django/utils/text.py

```diff
@@ -402,7 +402,7 @@ def slugify(value, allow_unicode=False):
         value = unicodedata.normalize('NFKC', value)
     else:
         value = unicodedata.normalize('NFKD', value).encode('ascii', 'ignore').decode('ascii')
-    value = re.sub(r'[^\w\s-]', '', value.lower()).strip()
+    value = re.sub(r'[^\w\s-]', '', value.lower()).strip('-_ ')
```

Comment:
`strip()` removes also other whitespace, e.g. `\t`. I think we should move strip to the next line, i.e.
```python
value = re.sub(r'[^\w\s-]', '', value.lower())
return re.sub(r'[-\s]+', '-', value).strip('-_')
```

Reply:
Just had a thought, should we strip all punctuation?

Reply:
We've already removed them in `re.sub(r'[^\w\s-]', '', value.lower())`.